run/SU2_preCICE_CHT.py

```diff
@@ -195,24 +194,22 @@ def main():
     comm.Barrier()
 
 
-  while (interface.is_coupling_ongoing()):
+  while (participant.is_coupling_ongoing()):
 
     # Implicit coupling
-    if (interface.is_action_required(precice.action_write_iteration_checkpoint())):
+    if (participant.requires_writing_checkpoint()):
       # Save the state
       SU2Driver.SaveOldState()
```

Comment:
That's what I understand:

https://github.com/precice/su2-adapter/blob/6540312608ed1c7e7b690fc48a51ebb3158b4b0c/replacement_files/python_wrapper_structure.cpp#L560-L562

@j-signorelli can confirm.

Reply:
The following part from below looks like it needs to go into the checkpointing:
```
TimeIter += 1
time += deltaT
```

I would imagine something like:
```
if (participant.requires_writing_checkpoint()):
    SU2Driver.SaveOldState()
    timeIterCp = TimeIter
    timeCp = time

....

if (participant.requires_reading_checkpoint()):
    # Reload old state
    SU2Driver.ReloadOldState()
    TimeIter = timeIterCp
    time = timeCp
```
This would also allow you to completely drop the `else`-part of the `if (participant.requires_reading_checkpoint()):` and just always increment `time` and `TimeIter` directly after calling `advance` (since this concludes the time step).

Reply:
As far as I know, these are just two different variations of the same (I recently saw it written somewhere):

- Always keep checkpoints, always advance the time, restore checkpoints if needed
- Do not advance the time by default, advance only when the iterations are complete.

The latter is only possible for some codes, and it is more efficient. In this case, it is supported and implemented in the adapter.

I tried converting to the first approach, but I already faced some issues advancing the output. I have the impression that the current solution works correctly, so I would prefer to not touch it, since I am not confident with SU2.

Reply:
#34 already addressed this.

Reply:
As a follow up I created a PR: https://github.com/precice/su2-adapter/pull/36

I think this sufficiently documents the open problems. All of this is not critical. We can just wait with #36 until after the release.